code/modules/food_and_drinks/food/snacks_other.dm

```diff
@@ -438,6 +445,47 @@
 	throw_speed = 1
 	throwforce = 0
 
+/obj/item/reagent_containers/food/snacks/lollipop/Destroy()
+	if(timer_id)
+		deltimer(timer_id)
+	..()
+
+/obj/item/reagent_containers/food/snacks/lollipop/equipped(mob/user, slot)
+	. = ..()
+	if(timer_id)
+		deltimer(timer_id)
+		timer_id = null
+	chewing = (slot == ITEM_SLOT_MASK ? TRUE : FALSE)
+	if(chewing) //Set a timer to chew(), instead of calling chew for the convenience of being able to equip/unequip our pop
+		timer_id = addtimer(CALLBACK(src, .proc/chew), bite_frequency, TIMER_STOPPABLE)	
```

Comment:
You need a dropped() for this too, for when items are dropped straight to the floor (stripping)